plugins/tiletypes.cpp

```diff
@@ -53,6 +56,25 @@ using namespace df::enums;
 DFHACK_PLUGIN("tiletypes");
 REQUIRE_GLOBAL(world);
 
+struct tiletypes_options {
+    // whether to display help
+    bool help = false;
+    bool quiet = false;
+
+    // if set, then use this position instead of the active game cursor
+    df::coord cursor;
+
+    static struct_identity _identity;
+};
+static const struct_field_info tiletypes_options_fields[] = {
+    { struct_field_info::PRIMITIVE, "help",   offsetof(tiletypes_options, help),   &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "quiet",  offsetof(tiletypes_options, quiet),  &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::SUBSTRUCT, "cursor", offsetof(tiletypes_options, cursor), &df::coord::_identity,                0, 0 },
+    { struct_field_info::END }
```

Comment:
Reminds me of https://github.com/DFHack/dfhack/pull/1849#discussion_r630710706 - we'll want to keep options in mind to avoid defining these manually in the long term.

Reply:
c++17 has some static reflection capabilities, such as iterating over fields. I have no experience with them, but the brief internet research I did indicates that we might be able to use it to simplify the identity struct generation. Is changing our target C++ standard an option?

I also wonder if we could do something in the current c++11 with templated functions (with specializations for primitive types). That's a little vague, but I could spend some time exploring that idea.

Reply:
We're unfortunately tied to the MSVC 2015 compiler on Windows, so even C++14 support is not great. (And with fancy C++ features come fancy error messages, slower compilation times, etc... although simplicity on our end would be good.)

I guess my vision is to have plugins be able to define their own types (in XML?) and have codegen produce metadata for them. It would be complicated to get everything linked in the right libraries, but we do something sort of similar with plugin-specific protobufs.